src/mapdata.h

```diff
@@ -344,28 +345,33 @@ struct submap {
     }
 
     inline void set_furn(int x, int y, furn_id furn) {
+        is_uniform = false;
         frn[x][y] = furn;
     }
 
     inline void set_ter(int x, int y, ter_id terr) {
+        is_uniform = false;
         ter[x][y] = terr;
     }
 
-    int get_radiation(int x, int y) {
+    inline int get_radiation(int x, int y) const {
         return rad[x][y];
     }
 
     void set_radiation(int x, int y, int radiation) {
+        is_uniform = false;
         rad[x][y] = radiation;
     }
 
     void update_lum_add(item const &i, int const x, int const y) {
+        is_uniform = false;
         if (i.is_emissive() && lum[x][y] < 255) {
             lum[x][y]++;
         }
     }
 
     void update_lum_rem(item const &i, int const x, int const y) {
+        is_uniform = false;
```

Comment:
AFAIK `lum` is not stored in the save, I understand that `is_uniform` only affects the decision to save the map / not save it, so a different `lum` value should not change `is_uniform`. Adding a (light emitting) item to the submap will trigger `is_uniform = false;` anyway.